obspy/io/sac/util.py

```diff
@@ -159,19 +159,24 @@ def is_same_byteorder(bo1, bo2):
     return (bo1.lower() in le) == (bo2.lower() in le)
 
 
-def _clean_str(value):
+# This fn is used in two places: in SACTrace.read, to sanitize strings for
+# SACTrace, and in sac_to_obspy_header, to sanitize strings for Trace that the
+# user may have manually added.
+def _clean_str(value, strip_whitespace=True):
     # Remove null values and whitespace, return a str
     try:
         # value is a str
         null_term = value.find('\x00')
     except TypeError:
         # value is a bytes
-        # null_term = value.decode().find('\x00')
         null_term = value.find(b'\x00')
+    except UnicodeError:
+        null_term = value.decode('ASCII', 'replace').find(b'\x00')
 
     if null_term >= 0:
-        value = value[:null_term]
-    value = value.strip()
+        value = value[:null_term] + b" " * len(value[null_term:])
```

Comment:
Changing `b' '` to `' '` or `u' '` or `native_str(' ')` fails tests on Python 3, with `TypeError: can't concat bytes to str`.  

 `_clean_str` probably needs someone more experienced than I am to look it over in more depth, particularly if we want any such changes in `1.0.3`, but at the moment every use case we've thrown at it in the test suite is currently passing.  I'm happy to open an issue for this after `1.0.3`.